flink-runtime/src/main/java/org/apache/flink/runtime/util/config/memory/jobmanager/JobManagerFlinkMemoryUtils.java

```diff
@@ -41,9 +41,8 @@ public JobManagerFlinkMemory deriveFromRequiredFineGrainedOptions(Configuration
 		MemorySize derivedTotalFlinkMemorySize = jvmHeapMemorySize.add(offHeapMemorySize);
 
 		if (config.contains(JobManagerOptions.TOTAL_FLINK_MEMORY)) {
-			// derive network memory from total flink memory, and check against network min/max
 			MemorySize totalFlinkMemorySize = ProcessMemoryUtils.getMemorySizeFromConfig(config, JobManagerOptions.TOTAL_FLINK_MEMORY);
-			if (derivedTotalFlinkMemorySize.getBytes() != totalFlinkMemorySize.getBytes()) {
+			if (derivedTotalFlinkMemorySize.getBytes() > totalFlinkMemorySize.getBytes()) {
```

Comment:
The check is correct in this case, the message is wrong, it should `does not match` instead of `exceeds`.
This has been fixed in https://github.com/apache/flink/pull/12557/files#diff-4ccfda6492f20dce963a7fa12cc240ebR65.
Therefore, I suggest to close this PR for now.